app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java

```diff
@@ -0,0 +1,189 @@
+        mSectionResourceId = sectionResourceId;
+        mTextResourceId = textResourceId;
+        mBaseAdapter = baseAdapter;
+        mContext = context;
+
+        mBaseAdapter.registerAdapterDataObserver(new RecyclerView.AdapterDataObserver() {
+            @Override
+            public void onChanged() {
+                mValid = mBaseAdapter.getItemCount()>0;
+                notifyDataSetChanged();
+            }
+
+            @Override
+            public void onItemRangeChanged(int positionStart, int itemCount) {
+                mValid = mBaseAdapter.getItemCount()>0;
+                notifyItemRangeChanged(positionStart, itemCount);
+            }
+
+            @Override
+            public void onItemRangeInserted(int positionStart, int itemCount) {
+                mValid = mBaseAdapter.getItemCount()>0;
+                notifyItemRangeInserted(positionStart, itemCount);
+            }
+
+            @Override
+            public void onItemRangeRemoved(int positionStart, int itemCount) {
+                mValid = mBaseAdapter.getItemCount()>0;
+                notifyItemRangeRemoved(positionStart, itemCount);
+            }
+        });
+    }
+
+
+    public static class SectionViewHolder extends RecyclerView.ViewHolder {
+
+        public TextView title;
+
+        public SectionViewHolder(View view, int mTextResourceid) {
+            super(view);
+            title = (TextView) view.findViewById(mTextResourceid);
+        }
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int typeView) {
+        if (typeView == SECTION_TYPE) {
+            final View view = LayoutInflater.from(mContext).inflate(mSectionResourceId, parent, false);
+            return new SectionViewHolder(view,mTextResourceId);
+        }else{
+            return mBaseAdapter.onCreateViewHolder(parent, typeView -1);
+        }
+    }
+
+
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder sectionViewHolder, int position) {
+        if (isSectionHeaderPosition(position)) {
+            ((SectionViewHolder)sectionViewHolder).title.setText(mSections.get(position).title);
+        }else{
+            mBaseAdapter.onBindViewHolder(sectionViewHolder,sectionedPositionToPosition(position));
+        }
+
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return isSectionHeaderPosition(position)
+                ? SECTION_TYPE
+                : mBaseAdapter.getItemViewType(sectionedPositionToPosition(position)) +1 ;
+    }
+
+
+    public static class Section {
+        int firstPosition;
+        int sectionedPosition;
+        CharSequence title;
+
+        public Section(int firstPosition, CharSequence title) {
+            this.firstPosition = firstPosition;
+            this.title = title;
+        }
+
+        public CharSequence getTitle() {
+            return title;
```

Comment:
return title is correct, there is no need of _this_ here. we are already seeing the value of title inside to a constructor.